tests/components/sensor/test_hydroquebec.py

```diff
@@ -0,0 +1,120 @@
+"""The test for the hydroquebec sensor platform."""
+import asyncio
+import sys
+from unittest.mock import MagicMock
+
+
+from homeassistant.bootstrap import async_setup_component
+from homeassistant.components.sensor import hydroquebec
+from tests.common import assert_setup_component
+
+
+CONTRACT = "123456789"
+
+
+class HydroQuebecClientMock(MagicMock):
+    """Fake Hydroquebec client."""
+
+    def __init__(self, username, password, contract=None):
+        """Fake Hydroquebec client init."""
+        pass
+
+    def get_data(self, contract):
+        """Return fake hydroquebec data."""
+        return {CONTRACT: {"balance": 160.12}}
+
+    def get_contracts(self):
+        """Return fake hydroquebec contracts."""
+        return [CONTRACT]
+
+    @asyncio.coroutine
+    def fetch_data(self):
+        """Return fake fetching data."""
+        pass
+
+
+class HydroQuebecClientMockError(MagicMock):
+    """Fake Hydroquebec client error."""
+
+    def get_data(self, contract):
+        """Return fake hydroquebec data."""
+        return {CONTRACT: {"balance": 160.12}}
+
+    @asyncio.coroutine
+    def get_contracts(self):
+        """Return fake hydroquebec contracts."""
+        raise PyHydroQuebecErrorMock("Fake Error")
+
+    @asyncio.coroutine
+    def fetch_data(self):
+        """Return fake fetching data."""
+        if hasattr(self, 'data_fetched'):
+            raise PyHydroQuebecErrorMock("Fake Error")
```

Comment:
Don't do this here. Just patch the call that should error, and add a side effect that raises the error. Tests should also be totally isolated from each other.